Metalama.Framework.Engine/Linking/LinkerInjectionStep.Rewriter.cs

```diff
@@ -504,35 +536,108 @@ void AddInjectionsOnPosition( InsertPosition position )
             }
         }
 
-        private static MemberDeclarationSyntax InjectStatementsIntoMemberDeclaration( IReadOnlyList<StatementSyntax> entryStatements, MemberDeclarationSyntax currentNode )
+        private static MemberDeclarationSyntax InjectStatementsIntoMemberDeclaration(
+            IMemberOrNamedType contextDeclaration,
+            IReadOnlyList<StatementSyntax> entryStatements,
+            MemberDeclarationSyntax currentNode )
         {
-            if (entryStatements.Count == 0)
+            if ( entryStatements.Count == 0 )
             {
                 return currentNode;
             }
 
             switch ( currentNode )
             {
                 case ConstructorDeclarationSyntax { Body: { } body } constructor:
-                    return constructor.WithBody( ReplaceBlock( entryStatements, body ) );
+                    return constructor.PartialUpdate( body: ReplaceBlock( entryStatements, body ) );
```

Comment:
Note that `PartialUpdate` with a single change will have slightly worse performance than a `With` method (which is why this change shouldn't be recommended by the analyzer). Though both approaches will allocate the same, which is the most important part.